packages/core/http/core-http-router-server-internal/src/versioned_router/handler_resolvers.ts

```diff
@@ -6,19 +6,38 @@
  * Side Public License, v 1.
  */
 
+/**
+ * Sort Kibana HTTP API versions from oldest to newest
+ *
+ * @example Given 'internal' versions ["1", "10", "2"] it will return ["1", "2", "10]
+ * @example Given 'public' versions ["2023-01-01", "2002-10-10", "2005-01-01"] it will return ["2002-10-10", "2005-01-01", "2023-01-01"]
+ */
+export const sort = (versions: string[], access: 'public' | 'internal') => {
```

Comment:
Sort of related to this PR, @pgayvallet found a bug in this logic that was not treating internal versions correctly. Updated and tried to preserve existing APIs as far as possible.